lib/api/src/grpc/transport_channel_pool.rs

```diff
@@ -1,30 +1,84 @@
 use std::collections::HashMap;
-use std::time::Duration;
+use std::future::Future;
+use std::num::NonZeroUsize;
+use std::time::{Duration, Instant};
 use std::vec::Vec;
 
 use rand::seq::SliceRandom;
-use tonic::transport::{Channel, Error, Uri};
+use tonic::transport::{Channel, Error as TonicError, Uri};
+use tonic::{Code, Status};
+
+const DEFAULT_GRPC_TIMEOUT: Duration = Duration::from_secs(10);
+const DEFAULT_POOL_SIZE: usize = 1;
+const CHANNEL_TTL: Duration = Duration::from_secs(5);
+
+struct ChannelsToUri {
```

Comment:
I find the name `ChannelsToUri` slightly confusing as it sounds like a lookup structure to lookup a channel by URI.

Alternative names could be:
- `PooledChannels`
- `ChannelGroup`
- `ConnectionPool`